src/core/services/theming/theming.js

```diff
@@ -528,7 +528,7 @@ function generateAllThemes($injector) {
   if (generateOnDemand) return;
 
   angular.forEach(THEMES, function(theme) {
-    if (!GENERATED[theme.name]) {
+    if (!GENERATED[theme.name] && !(theme.name !== $mdTheming.defaultTheme() && theme.name === 'default')) {
```

Comment:
Oh it makes sense. An `if-expression` won't continue the check if it already failed. 
For example:
```js
true && false && true
-> The check will immediately cancel itself, because the second `AND` is false.
```

The theme should be skipped if the theme name is equal to  `default` and if the default theme has been changed. 

But maybe we can find a more elegant check, but your logic in the check isn't valid.